tools/cp-utility/src/main.rs

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+use std::collections::VecDeque;
+use std::env;
+use std::fs;
+use std::io;
+use std::os::unix;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process;
+
+/// A type of copy operation
+#[derive(Debug, PartialEq)]
+enum CopyType {
+    /// equivalent to cp <source> <dest>
+    SingleFile,
+    /// equivalent to cp -a <source> <dest>
+    Archive,
+}
+
+/// Encapsulate a copy operation
+struct CopyOperation {
+    /// The source path
+    source: PathBuf,
+    /// The destination path
+    destination: PathBuf,
+    /// The type of copy being performed
+    copy_type: CopyType,
+}
+
+/// Parse command line arguments and transform into `CopyOperation`
+fn parse_args(args: Vec<&str>) -> io::Result<CopyOperation> {
+    if !((args.len() == 4 || args.len() == 5 && args[2].eq("-a")) && args[1].eq("cp")) {
```

Comment:
Should we expand this error checking? Verify that `args[1] == "cp"` for both arg lengths?

Reply:
oh. it is already checking.
this is the condition that is being checked in negative `(args.len() == 4 || args.len() == 5 && args[2].eq("-a")) && args[1].eq("cp")`

note the `&& args[2].eq("-a")` is considering both the cases of 4 and 5 arguments.

Reply:
Ahh good points. I missed the nested parenthesis.

Reply:
Can we add a verification test case?

Reply:
good call. I added more tests.

Reply:
https://github.com/aws-observability/aws-otel-java-instrumentation/pull/353/files#diff-255e1973e4371a91e310ea9c1b94c1507bab752b6aa20b39322542e6d205616aR178-R179